container/lxd/lxd.go

```diff
@@ -196,6 +212,41 @@ func (manager *containerManager) CreateContainer(
 	return
 }
 
+// getImageSources returns a list of LXD remote image sources based on the
+// configuration that was passed into the container manager.
+func (manager *containerManager) getImageSources() ([]lxdclient.Remote, error) {
+	imURL := manager.imageMetadataURL
+
+	// Unless the configuration explicitly requests the daily stream,
+	// an empty image metadata URL results in a search of the default sources.
+	if imURL == "" && manager.imageStream != "daily" {
+		logger.Debugf("checking default image metadata sources")
+		return lxdclient.DefaultImageSources, nil
+	}
+	// Otherwise only check the daily stream.
+	if imURL == "" {
+		return lxdclient.DefaultImageSources[1:], nil
```

Comment:
this feels a little wonky, as it presumes the exact content of DefaultImageSources which is being imported from another package. It presumes both the order and the implication of those values.
I'd be ok with landing this in 2.3 as long as we have plans to change it as we rewrite the work in 2.4 as we try to get rid of lxdclient anyway.
Ideally we would just track sources as being 'released' or 'daily' and then just package up "start here, and fallback to there" for whatever explicit reasons.

Reply:
This is addressed. Sources are included explicitly and _DefaultImageSources_ is removed.